DataMigration/src/main/java/org/sagebionetworks/dian/datamigration/MigrationUtil.java

```diff
@@ -0,0 +1,438 @@
+        }
+        return pathMap;
+    }
+
+    /**
+     * @param participantId raw participant ID from HM data model file
+     *                      this may be any length 0 to 6
+     * @return the participant ID with a length of 6, with leading "0"s added
+     */
+    public static String fixParticipantId(String participantId) {
+        if (participantId.length() >= PARTICIPANT_ID_LENGTH) {
+            return participantId.substring(0, PARTICIPANT_ID_LENGTH);
+        }
+        int zerosToAdd = PARTICIPANT_ID_LENGTH - participantId.length();
+        String zerosPrefix = StringUtils.repeat("0", zerosToAdd);
+        return zerosPrefix + participantId;
+    }
+
+    /**
+     * @param participantJsonFolder each file in this list represents a directory
+     *                              where the participant JSON files were unzipped to
+     *                              these should map to SynapseUtil.DownloadFolder.userFolders()
+     * @return the full list of hasd and exr users
+     * @throws IOException if something goes wrong
+     */
+    public static @NonNull List<HmUser> createHmUserRaterData(
+            List<Path> participantJsonFolder) throws IOException {
+
+        List<HmUser> data = new ArrayList<>();
+        ObjectMapper mapper = new ObjectMapper();
+
+        for (Path folder : participantJsonFolder) {
+
+            List<HmUser> userList = new ArrayList<>();
+            Map<ParticipantFileEnum, Path> pathList = MigrationUtil.findParticipantPaths(folder);
+
+            Participant[] participantList = TableRow.parseTableRow(mapper, pathList.get(
+                    ParticipantFileEnum.PARTICIPANT), Participant[].class);
+            Rater[] raters = TableRow.parseTableRow(mapper, pathList.get(
+                    ParticipantFileEnum.RATER), Rater[].class);
+            SiteLocation[] siteLocList = TableRow.parseTableRow(mapper, pathList.get(
+                    ParticipantFileEnum.SITE_LOCATION), SiteLocation[].class);
+            ParticipantPhone[] phoneList = TableRow.parseTableRow(mapper, pathList.get(
+                    ParticipantFileEnum.PHONE), ParticipantPhone[].class);
+            ParticipantNotes[] participantNotes = TableRow.parseTableRow(mapper, pathList.get(
+                    ParticipantFileEnum.PARTICIPANT_NOTES), ParticipantNotes[].class);
+            ParticipantDeviceId[] participantDeviceIds = TableRow.parseTableRow(mapper, pathList.get(
+                    ParticipantFileEnum.PARTICIPANT_DEVICE_ID), ParticipantDeviceId[].class);
+            ParticipantRater[] participantRater = TableRow.parseTableRow(mapper, pathList.get(
+                    ParticipantFileEnum.PARTICIPANT_RATER), ParticipantRater[].class);
+            ParticipantSiteLocation[] participantSiteLocList = TableRow.parseTableRow(mapper, pathList.get(
+                    ParticipantFileEnum.PARTICIPANT_SITE_LOCATION), ParticipantSiteLocation[].class);
+
+            for (Participant user : participantList) {
```

Comment:
Usually I would agree here and do a null check.  However, if there are no participants created, there is something wrong either with the export from HappyMedium or my migration code, and I would prefer this to crash. Let me know if you feel strongly otherwise.